doc/catalog-format-spec.rst

```diff
@@ -285,6 +285,32 @@ entries:
    linux   = { OS = "gnu-linux" } # Compact table syntax is convenient in this case
    windows = { OS = "ms-linux" }  # to see all enumeration values, one per row.
 
+* ``environment``: optional table used to modify environment variables that
+  will apply at build time. Variables and values are specified with the form
+  ``VARIABLE.<action> = "value"``, where ``<action>`` is one of ``append``,
+  ``prepend``, or ``set``. For instance:
+
+  .. code-block:: toml
+
+   [environment]
+   C_INCLUDE_PATH.append = "/usr/include/something"
+   PATH.prepend = "$CRATE_ROOT/bin"
+   MSYS.set = "$MSYS_ROOT"
+
+  Other environment variables may appear in the value, and will be replaced.
+  Furthermore, a few predefined variables are provided by Alire:
+
+  * ``$CRATE_ROOT`` is the deployment directory of the crate.
```

Comment:
We should discuss the predefined value format. I think this `${CRATE}` will be easier to handle and less error prone.

Reply:
Sure. At the moment there's no code about that anyway, so it's a matter of expressing what we want in the documentation. I too like better with `{}`.